ExtensionSpec: The ActorSystem extensions support should fail the actor system if an extension listed in akka.extensions fails to start

https://jenkins.akka.io:8498/job/pr-validator-per-commit-jenkins/18260/consoleFull

```
[11-26 18:11:15.486] [info]   Expected exception java.lang.RuntimeException to be thrown, but no exception was thrown (ExtensionSpec.scala:88)
[11-26 18:11:15.487] [info]   org.scalatest.exceptions.TestFailedException:
[11-26 18:11:15.487] [info]   at org.scalatest.Assertions.newAssertionFailedException(Assertions.scala:530)
[11-26 18:11:15.487] [info]   at org.scalatest.Assertions.newAssertionFailedException$(Assertions.scala:529)
[11-26 18:11:15.487] [info]   at org.scalatest.WordSpec.newAssertionFailedException(WordSpec.scala:1881)
[11-26 18:11:15.487] [info]   at org.scalatest.Assertions.intercept(Assertions.scala:824)
[11-26 18:11:15.487] [info]   at org.scalatest.Assertions.intercept$(Assertions.scala:804)
[11-26 18:11:15.487] [info]   at org.scalatest.WordSpec.intercept(WordSpec.scala:1881)
[11-26 18:11:15.487] [info]   at akka.actor.ExtensionSpec.$anonfun$new$6(ExtensionSpec.scala:88)
[11-26 18:11:15.487] [info]   at org.scalatest.OutcomeOf.outcomeOf(OutcomeOf.scala:85)
[11-26 18:11:15.487] [info]   at org.scalatest.OutcomeOf.outcomeOf$(OutcomeOf.scala:83)
[11-26 18:11:15.487] [info]   at org.scalatest.OutcomeOf$.outcomeOf(OutcomeOf.scala:104)
[11-26 18:11:15.487] [info]   at org.scalatest.Transformer.apply(Transformer.scala:22)
[11-26 18:11:15.487] [info]   at org.scalatest.Transformer.apply(Transformer.scala:20)
[11-26 18:11:15.487] [info]   at org.scalatest.WordSpecLike$$anon$3.apply(WordSpecLike.scala:1075)
[11-26 18:11:15.487] [info]   at org.scalatest.TestSuite.withFixture(TestSuite.scala:196)
[11-26 18:11:15.487] [info]   at org.scalatest.TestSuite.withFixture$(TestSuite.scala:195)
[11-26 18:11:15.487] [info]   at org.scalatest.WordSpec.withFixture(WordSpec.scala:1881)
[11-26 18:11:15.487] [info]   at org.scalatest.WordSpecLike.invokeWithFixture$1(WordSpecLike.scala:1073)
[11-26 18:11:15.488] [info]   at org.scalatest.WordSpecLike.$anonfun$runTest$1(WordSpecLike.scala:1085)
[11-26 18:11:15.488] [info]   at org.scalatest.SuperEngine.runTestImpl(Engine.scala:286)
[11-26 18:11:15.488] [info]   at org.scalatest.WordSpecLike.runTest(WordSpecLike.scala:1085)
[11-26 18:11:15.488] [info]   at org.scalatest.WordSpecLike.runTest$(WordSpecLike.scala:1067)
[11-26 18:11:15.488] [info]   at org.scalatest.WordSpec.runTest(WordSpec.scala:1881)
[11-26 18:11:15.488] [info]   at org.scalatest.WordSpecLike.$anonfun$runTests$1(WordSpecLike.scala:1144)
[11-26 18:11:15.488] [info]   at org.scalatest.SuperEngine.$anonfun$runTestsInBranch$1(Engine.scala:393)
[11-26 18:11:15.488] [info]   at scala.collection.immutable.List.foreach(List.scala:392)
[11-26 18:11:15.488] [info]   at org.scalatest.SuperEngine.traverseSubNodes$1(Engine.scala:381)
[11-26 18:11:15.488] [info]   at org.scalatest.SuperEngine.runTestsInBranch(Engine.scala:370)
[11-26 18:11:15.488] [info]   at org.scalatest.SuperEngine.$anonfun$runTestsInBranch$1(Engine.scala:407)
[11-26 18:11:15.488] [info]   at scala.collection.immutable.List.foreach(List.scala:392)
[11-26 18:11:15.488] [info]   at org.scalatest.SuperEngine.traverseSubNodes$1(Engine.scala:381)
[11-26 18:11:15.488] [info]   at org.scalatest.SuperEngine.runTestsInBranch(Engine.scala:376)
[11-26 18:11:15.488] [info]   at org.scalatest.SuperEngine.runTestsImpl(Engine.scala:458)
[11-26 18:11:15.488] [info]   at org.scalatest.WordSpecLike.runTests(WordSpecLike.scala:1144)
[11-26 18:11:15.488] [info]   at org.scalatest.WordSpecLike.runTests$(WordSpecLike.scala:1143)
[11-26 18:11:15.488] [info]   at org.scalatest.WordSpec.runTests(WordSpec.scala:1881)
[11-26 18:11:15.488] [info]   at org.scalatest.Suite.run(Suite.scala:1124)
[11-26 18:11:15.489] [info]   at org.scalatest.Suite.run$(Suite.scala:1106)
[11-26 18:11:15.489] [info]   at org.scalatest.WordSpec.org$scalatest$WordSpecLike$$super$run(WordSpec.scala:1881)
[11-26 18:11:15.489] [info]   at org.scalatest.WordSpecLike.$anonfun$run$1(WordSpecLike.scala:1189)
[11-26 18:11:15.489] [info]   at org.scalatest.SuperEngine.runImpl(Engine.scala:518)
[11-26 18:11:15.489] [info]   at org.scalatest.WordSpecLike.run(WordSpecLike.scala:1189)
[11-26 18:11:15.489] [info]   at org.scalatest.WordSpecLike.run$(WordSpecLike.scala:1187)
[11-26 18:11:15.489] [info]   at org.scalatest.WordSpec.run(WordSpec.scala:1881)
[11-26 18:11:15.489] [info]   at org.scalatest.tools.Framework.org$scalatest$tools$Framework$$runSuite(Framework.scala:317)
[11-26 18:11:15.489] [info]   at org.scalatest.tools.Framework$ScalaTestTask.execute(Framework.scala:510)
[11-26 18:11:15.489] [info]   at sbt.ForkMain$Run.lambda$runTest$1(ForkMain.java:304)
[11-26 18:11:15.489] [info]   at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[11-26 18:11:15.489] [info]   at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[11-26 18:11:15.489] [info]   at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[11-26 18:11:15.489] [info]   at java.lang.Thread.run(Thread.java:748)
[ERROR] [11/26/2019 19:11:15.525] [pool-1-thread-1-ScalaTest-running-ExtensionSpec] [akka.actor.ActorSystemImpl(failing)] While trying to load extension [akka.actor.MissingExtension], skipping...
java.lang.ClassNotFoundException: akka.actor.MissingExtension
	at java.net.URLClassLoader.findClass(URLClassLoader.java:382)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:349)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	at java.lang.Class.forName0(Native Method)
	at java.lang.Class.forName(Class.java:348)
	at akka.actor.ReflectiveDynamicAccess.$anonfun$getClassFor$1(ReflectiveDynamicAccess.scala:29)
	at scala.util.Try$.apply(Try.scala:213)
	at akka.actor.ReflectiveDynamicAccess.getClassFor(ReflectiveDynamicAccess.scala:28)
	at akka.actor.ReflectiveDynamicAccess.createInstanceFor(ReflectiveDynamicAccess.scala:47)
	at akka.actor.ActorSystemImpl$$anonfun$1.applyOrElse(ActorSystem.scala:1192)
	at akka.actor.ActorSystemImpl$$anonfun$1.applyOrElse(ActorSystem.scala:1191)
	at scala.runtime.AbstractPartialFunction.apply(AbstractPartialFunction.scala:38)
	at scala.util.Failure.recoverWith(Try.scala:236)
	at akka.actor.ActorSystemImpl.$anonfun$loadExtensions$1(ActorSystem.scala:1191)
	at scala.collection.Iterator.foreach(Iterator.scala:941)
	at scala.collection.Iterator.foreach$(Iterator.scala:941)
	at scala.collection.AbstractIterator.foreach(Iterator.scala:1429)
	at scala.collection.IterableLike.foreach(IterableLike.scala:74)
	at scala.collection.IterableLike.foreach$(IterableLike.scala:73)
	at scala.collection.AbstractIterable.foreach(Iterable.scala:56)
	at akka.actor.ActorSystemImpl.loadExtensions$1(ActorSystem.scala:1190)
	at akka.actor.ActorSystemImpl.loadExtensions(ActorSystem.scala:1210)
	at akka.actor.ActorSystemImpl.liftedTree2$1(ActorSystem.scala:1042)
	at akka.actor.ActorSystemImpl._start$lzycompute(ActorSystem.scala:1029)
	at akka.actor.ActorSystemImpl._start(ActorSystem.scala:1029)
	at akka.actor.ActorSystemImpl.start(ActorSystem.scala:1052)
	at akka.actor.ActorSystem$.apply(ActorSystem.scala:282)
	at akka.actor.ActorSystem$.apply(ActorSystem.scala:326)
	at akka.actor.ActorSystem$.apply(ActorSystem.scala:300)
	at akka.actor.ExtensionSpec.$anonfun$new$8(ExtensionSpec.scala:102)
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
	at org.scalatest.OutcomeOf.outcomeOf(OutcomeOf.scala:85)
	at org.scalatest.OutcomeOf.outcomeOf$(OutcomeOf.scala:83)
	at org.scalatest.OutcomeOf$.outcomeOf(OutcomeOf.scala:104)
	at org.scalatest.Transformer.apply(Transformer.scala:22)
	at org.scalatest.Transformer.apply(Transformer.scala:20)
	at org.scalatest.WordSpecLike$$anon$3.apply(WordSpecLike.scala:1075)
	at org.scalatest.TestSuite.withFixture(TestSuite.scala:196)
	at org.scalatest.TestSuite.withFixture$(TestSuite.scala:195)
	at org.scalatest.WordSpec.withFixture(WordSpec.scala:1881)
	at org.scalatest.WordSpecLike.invokeWithFixture$1(WordSpecLike.scala:1073)
	at org.scalatest.WordSpecLike.$anonfun$runTest$1(WordSpecLike.scala:1085)
	at org.scalatest.SuperEngine.runTestImpl(Engine.scala:286)
	at org.scalatest.WordSpecLike.runTest(WordSpecLike.scala:1085)
	at org.scalatest.WordSpecLike.runTest$(WordSpecLike.scala:1067)
	at org.scalatest.WordSpec.runTest(WordSpec.scala:1881)
	at org.scalatest.WordSpecLike.$anonfun$runTests$1(WordSpecLike.scala:1144)
	at org.scalatest.SuperEngine.$anonfun$runTestsInBranch$1(Engine.scala:393)
	at scala.collection.immutable.List.foreach(List.scala:392)
	at org.scalatest.SuperEngine.traverseSubNodes$1(Engine.scala:381)
	at org.scalatest.SuperEngine.runTestsInBranch(Engine.scala:370)
	at org.scalatest.SuperEngine.$anonfun$runTestsInBranch$1(Engine.scala:407)
	at scala.collection.immutable.List.foreach(List.scala:392)
	at org.scalatest.SuperEngine.traverseSubNodes$1(Engine.scala:381)
	at org.scalatest.SuperEngine.runTestsInBranch(Engine.scala:376)
	at org.scalatest.SuperEngine.runTestsImpl(Engine.scala:458)
	at org.scalatest.WordSpecLike.runTests(WordSpecLike.scala:1144)
	at org.scalatest.WordSpecLike.runTests$(WordSpecLike.scala:1143)
	at org.scalatest.WordSpec.runTests(WordSpec.scala:1881)
	at org.scalatest.Suite.run(Suite.scala:1124)
	at org.scalatest.Suite.run$(Suite.scala:1106)
	at org.scalatest.WordSpec.org$scalatest$WordSpecLike$$super$run(WordSpec.scala:1881)
	at org.scalatest.WordSpecLike.$anonfun$run$1(WordSpecLike.scala:1189)
	at org.scalatest.SuperEngine.runImpl(Engine.scala:518)
	at org.scalatest.WordSpecLike.run(WordSpecLike.scala:1189)
	at org.scalatest.WordSpecLike.run$(WordSpecLike.scala:1187)
	at org.scalatest.WordSpec.run(WordSpec.scala:1881)
	at org.scalatest.tools.Framework.org$scalatest$tools$Framework$$runSuite(Framework.scala:317)
	at org.scalatest.tools.Framework$ScalaTestTask.execute(Framework.scala:510)
	at sbt.ForkMain$Run.lambda$runTest$1(ForkMain.java:304)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

Perhaps related to #28248 ?

--- a/akka-actor/src/main/scala/akka/actor/ActorSystem.scala
+++ b/akka-actor/src/main/scala/akka/actor/ActorSystem.scala
@@ -1179,29 +1179,25 @@ private[akka] class ActorSystemImpl(
   private def loadExtensions(): Unit = {
 
     /*
-     * @param throwOnLoadFail Throw exception when an extension fails to load (needed for backwards compatibility)
+     * @param throwOnLoadFail
+     *  Throw exception when an extension fails to load (needed for backwards compatibility.
+     *    when the extension cannot be found at all we throw regardless of this setting)
      */
     def loadExtensions(key: String, throwOnLoadFail: Boolean): Unit = {
-      def handleException(fqcn: String, problem: Throwable): Unit = {
-        if (!throwOnLoadFail) log.error(problem, "While trying to load extension [{}], skipping...", fqcn)
-        else throw new RuntimeException(s"While trying to load extension [$fqcn]", problem)
-      }
-
       immutableSeq(settings.config.getStringList(key)).foreach { fqcn =>
         dynamicAccess.getObjectFor[AnyRef](fqcn).recoverWith {
           case _ => dynamicAccess.createInstanceFor[AnyRef](fqcn, Nil)
         } match {
           case Success(p: ExtensionIdProvider) =>
-            try registerExtension(p.lookup())
-            catch { case NonFatal(e) => handleException(fqcn, e) }
+            registerExtension(p.lookup())
           case Success(p: ExtensionId[_]) =>
-            try registerExtension(p)
-            catch { case NonFatal(e) => handleException(fqcn, e) }
+            registerExtension(p)
           case Success(_) =>
             if (!throwOnLoadFail) log.error("[{}] is not an 'ExtensionIdProvider' or 'ExtensionId', skipping...", fqcn)
             else throw new RuntimeException(s"[$fqcn] is not an 'ExtensionIdProvider' or 'ExtensionId'")
           case Failure(problem) =>
-            handleException(fqcn, problem)
+            if (!throwOnLoadFail) log.error(problem, "While trying to load extension [{}], skipping...", fqcn)
+            else throw new RuntimeException(s"While trying to load extension [$fqcn]", problem)
         }
       }
     }

